query/src/test/java/com/connexta/replication/query/cmp/QueryServiceComponentTest.java

```diff
@@ -0,0 +1,416 @@
+
+    assertThat(
+        "timed out waiting for expected published task",
+        queue.waitForPendingSizeToReach(2, 5L, TimeUnit.SECONDS),
+        Matchers.equalTo(true));
+
+    assertTask(queue.take(), metadataList.get(2), (byte) 9, OperationType.HARVEST);
+    assertTask(queue.take(), metadataList.get(0), (byte) 0, OperationType.HARVEST);
+
+    assertIndexSaved(filterList.get(0), metadataList.get(0).getMetadataModified());
+    assertIndexSaved(filterList.get(2), metadataList.get(2).getMetadataModified());
+  }
+
+  /*
+  Verifies that ION and Tactical DDF sites are handled correctly
+   */
+  @Test
+  public void testQueryMultipleSites() throws Exception {
+    List<Site> siteList =
+        List.of(
+            new SiteImpl(generateSitePojo(1).setKind(SiteKind.TACTICAL)),
+            new SiteImpl(generateSitePojo(2).setType(SiteType.ION)),
+            new SiteImpl(generateSitePojo(3)));
+    List<Filter> filterList =
+        List.of(
+            new FilterImpl(generateFilterPojo("1", 11)),
+            new FilterImpl(generateFilterPojo("2", 21)),
+            new FilterImpl(generateFilterPojo("3", 31)));
+    List<Metadata> metadataList =
+        List.of(
+            generateMetadataWithResource(111),
+            generateMetadataWithResource(211),
+            generateMetadataWithResource(311));
+
+    // for each site, save it in the siteManager and save the associated filter in the
+    // filterManager.
+    // when a query is made with a filter, return the associated metacard.
+    for (int i = 0; i < siteList.size(); i++) {
+      siteManager.save(siteList.get(i));
+      filterManager.save(filterList.get(i));
+      when(adapter.query(Mockito.argThat(isQueryRequestWithFilter(filterList.get(i).getFilter()))))
+          .thenReturn(new QueryResponseImpl(List.of(metadataList.get(i)), this::collectError));
+    }
+
+    startQueryManager(THIRTY_MINUTES);
+
+    // grab the queue for each site and verify only the queues for valid sites get tasks put in
+    // them.
+    MemorySiteQueue tacticalQueue = (MemorySiteQueue) broker.getQueue(siteList.get(0).getId());
+    MemorySiteQueue ionQueue = (MemorySiteQueue) broker.getQueue(siteList.get(1).getId());
+    MemorySiteQueue regionalQueue = (MemorySiteQueue) broker.getQueue(siteList.get(2).getId());
+
+    assertThat(
+        "timed out waiting for expected published task",
+        regionalQueue.waitForPendingSizeToReach(1, 5L, TimeUnit.SECONDS),
+        Matchers.equalTo(true));
+    assertTask(regionalQueue.take(), metadataList.get(2), (byte) 0, OperationType.HARVEST);
+    assertThat(tacticalQueue.pendingSize(), Matchers.equalTo(0));
+    assertThat(ionQueue.pendingSize(), Matchers.equalTo(0));
+
+    assertIndexSaved(filterList.get(2), metadataList.get(2).getMetadataModified());
+  }
+
+  private void assertIndexSaved(Filter filter, Date lastModified) {
+    Awaitility.await().until(() -> indexManager.getOrCreate(filter).getModifiedSince().isPresent());
```

Comment:
❗️ We might want to put a timeout here so that if it doesn't happen in 5-10 seconds, we fail the test case instead of letting it be stuck.